goosepaper/upload.py

```diff
@@ -17,51 +52,66 @@ def validateFolder(folder: str):
 
     return True
 
-def upload(filepath=None, replace=False, folder=None):
 
-    if not filepath:
-        parser = argparse.ArgumentParser("Upload Goosepaper to reMarkable tablet")
-        parser.add_argument(
-            "file", default=None, help="The file to upload",
-        )
-        args = parser.parse_args()
-        filepath = args.file
+def getallitems(client):
 
-    filepath = Path(filepath)
+    # So somehow I corrupted or broke my cloud during testing and any object
+    # which exists is getting returned twice in the object list from
+    # get_meta_items. Deleting items and re-adding them hasn't fixed it even
+    # using the official RM client. So this avoids that problem. I haven't found
+    # a real fix yet for my cloud though.
+    
+    allitems = [ item for item in client.get_meta_items() if (item.Parent != "trash") ]
 
-    client = Client()
+    items=[]
+    for tempitem in allitems:
+        if not any(item.ID == tempitem.ID for item in items):
+            items.append(tempitem)
+
+    return items
 
-    try:
-        client.renew_token()
-    except AuthError:
-        print(
-            "Looks like this if the first time you've uploaded, need to register the device"
-        )
-        print("Get the code from here: https://my.remarkable.com/connect/remarkable")
-        code = input()
-        print("registering")
-        client.register_device(code)
-        if not client.renew_token():
-            print("registration failed D:")
-        else:
-            print("registration successful")
+
+def upload(filepath, replace=False, folder=None):
+
+    client = auth_client()
+    
+    if not client: 
+        print ("Honk Honk! Couldn't auth! Is your rmapy configured?")
+        return False
 
     if not validateFolder(folder):
         return False
 
+    if not sanitycheck(folder,client):
```

Comment:
Could do that but there isn't a global goosepaper.prefs config yet (is there?). But since it's throwing an exception and exiting with an error it shouldn't overwrite anything....

Reply:
We have a system written by @traysh (!! 😍 ) that accepts config parameters either from the goosepaper config or from the command line; that might be a good home for it now, with the hope to later on meet the spec in #44!

Reply:
Okay then. Once you're merged the commits and I'm updated to match your master I'll add that in as a parameter passed to upload() and add it to the command line parser.

Reply:
I don't want to be a bother, but is it okay if we add it in this PR? I don't want to merge changes to master that might break existing workflows. (let me know if this is too annoying to add here!)

Reply:
So what I did was removed the actual check (left the function def in place) because I want to put that into the preferences config file option as well as the command line and it'll give a 'new feature' when the goosepaper-prefs.json (or whaever) is introduced. :)